doc/datatable.md

```diff
@@ -0,0 +1,350 @@
+	useEffect(() => {
+		fetchData(page);
+	}, [page]);
+
+	...
+
+	return (
+		<DataTable
+			...
+			data={data}
+			count={count}
+			currentPage={page}
+			setPage={setPage}
+			...
+		/>
+	)
+
+}
+```
+
+Prop `count` is ussualy obtained from API endpoint with fetched data.
+
+Example of fetched data:
+
+```
+{
+	data: [...],
+	count: /*some amount*/
+}
+```
+
+# Optional
+
+`DataTable` can also accept optional props `limit`, `setLimit`, `createButton`, `search` and `onSearch`.
+
+Example of `DataTable` with all props:
+
+```
+<DataTable
+	title={{text: "Table Demo", icon: 'cil-user'}}
+	data={data}
+	headers={headers}
+	count={count}
+	limit={limit}
+	setLimit={setLimit}
+	currentPage={page}
+	setPage={setPage}
+	search={{ icon: 'icon', placeholder: 'placeholder' }}
+	onSearch={onSearch}
+	createButton={{ text: "Create", icon: 'icon', pathname: '#' }}
+/>
+```
+
+Props `limit` and `setLimit` is used to handle limiting the amount of objects per page. Prop `limit` is number and represents the amount of rendered objects per page and calculate amount of pages in `DataTable`, default value is 10. Prop `setLimit` is a function for changing `limit`. If `setLimit` exists `DataTable` renders dropdown box for setting limit with values 5, 10, 15, 20.
+
+Example of handling and using:
+
+```
+import React, { useState } from 'react';
+
+...
+
+function (props) {
+	...
+	const [limit, setLimit] = useState(10);
+	...
+	return (
+		<DataTable
+			...
+			limit={limit}
+			setLimit={setLimit}
+			...
+		>
+	);
+}
+```
+
+Prop `createButton` is used for rendering button with a link for redirecting to create page. It is an object with obligatory properties `text` and `pathname` which are text inside the button and pathname for redirecting respectively.
+`createButton` also accepts optional property `icon` for rendering icon on the right side of the text inside the button.
+Property `icon` can either be a React Component (e.g. from package `@material-ui/icons`) or a string which is className and will be rendered as <i className={icon}></icon>
+
+Example:
+
+```
+	...
+	return (
+		<DataTable
+			...
+			createButton={{ text: "Create", icon: "icon-classname", pathname: "#" }}
+			...
+		>
+	);
+	...
+```
+
+`createButton` may also be obtained from configuration.
+
+Example of `createButton` in `configDefaults`:
+```
+let ConfigDefaults = {
+	...
+	table: {
+		create_button: { text: "Create", icon: "icon-classname", pathname: "#" }
+	},
+	...
+};
+```
+
+Props `search` and `onSearch` is used to render input search box in `DataTable`. First prop `search` is either a boolean or an object containing optional property `placeholder` for placeholder of search input box and optional property `icon` which can either be a React Component (e.g. from package `@material-ui/icons`) or a string which is className and will be rendered as <i className={icon}></icon>.Second prop `onSearch` is a function which executes after 500ms after last change in search input box has been made.
```

Comment:
@titkov49 onSearch will search only among the items on the current page, am I right? It might be good if you can at least mention here, that the search can be done also on axios request to the database. (with example)